packages/plugin-ext/src/plugin/terminal-ext.ts

```diff
@@ -189,7 +189,18 @@ export class TerminalServiceExtImpl implements TerminalServiceExt {
             terminal.deferredProcessId = new Deferred<number>();
             terminal.deferredProcessId.resolve(processId);
         }
-        const pseudoTerminal = this._pseudoTerminals.get(terminalId.toString());
+
+        // Switch the pseudoterminal keyed by terminalId to be keyed by terminal ID
+        const tId = terminalId.toString();
+        if (this._pseudoTerminals.has(tId)) {
+            const pseudo = this._pseudoTerminals.get(tId);
+            if (pseudo) {
```

Comment:
Does it ever make sense that the pseudo-terminal is `falsy` if the map `has` the key in question?

Reply:
No, but `Map.get()` returns `thing | undefined`, so won't compile without that guard IIRC. The only change I think can be made here is to remove the outer `has()` check. Want me to do that?

Reply:
OK